api/v1/services/CheckInService.js

```diff
@@ -69,42 +69,39 @@ module.exports.createCheckIn = function (attributes){
     return checkin.validate()
         .catch(CheckitError, utils.errors.handleValidationError)
         .then(function(validation) {
-            return CheckIn.findByUserId(attributes.userId);
+          return CheckIn.transaction(function (t) {
+              return checkin.save(null, {transacting: t})
+              .then(function(model){
```

Comment:
This is _way_ too nested. I realize it was originally written like this, but it's so hard to read. If there's a `return` inside of this top-level promise block, it should result in a new (non-nested) block being formed.